custom/inddex/couchdata.py

```diff
@@ -0,0 +1,150 @@
+from memoized import memoized
+
+from corehq.apps.fixtures.models import FixtureDataItem, FixtureDataType
+
+
+class CouchData:
```

Comment:
I am sure I'm missing a lot of context on what this class is meant to do. Could you rename it so that its name explains a little about its purpose? Also, could you add a docstring to explain why it does what it does?

There is a principle behind my request. The comments that you included in this PR show me that you understand this principle, but I'd really like you to carry it further.

Other people will need to maintain this code for years to come, and you won't necessarily be available to explain what it does, or why it does it. The code has to do both those things by itself.

For example, one day we will migrate Fixtures from Couch to Postgres, and the person responsible for that might never have seen this PR. It will be really important for them to know what this is doing and why. Please help us and future devs by seeing if there is a way to simplify how this works, and explain its purpose, and why it fulfils that purpose the way it does.